command/debug/debug.go

```diff
@@ -382,6 +400,63 @@ func (c *cmd) captureStatic() error {
 	return errs
 }
 
+func (c *cmd) troubleshootPorts() error {
+	output := troubleshootRun(default_ports, "localhost")
+
+	if err := writeJSONFile(filepath.Join(c.output, targetPorts+".json"), output); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func troubleshootRun(ports []string, host string) PortStatus {
+	openPorts := make(chan string, len(ports))
+	closePorts := make(chan string, len(ports))
+
+	var portResults PortStatus
+	var wg sync.WaitGroup
+	wg.Add(len(ports))
+
+	for _, port := range ports {
+		go func(port string) {
+			defer wg.Done()
+			address, _ := url.Parse(net.JoinHostPort(host, port))
+
+			err := dialPort(*address)
+			if err != nil {
+				closePorts <- port
+			} else {
+				// If no error occurs, the connection was successful, and the port is open.
+				openPorts <- port
+			}
+		}(port)
+	}
+
+	wg.Wait()
+	close(openPorts)
+	close(closePorts)
+
+	for port := range openPorts {
```

Comment:
Directly append instead of using for loop
```suggestion
	portResults.Open = append(portResults.Open, openPorts...)
```

Reply:
openPorts is a string channel, I don't think append can take that as an argument.